src/match_identifiers.cpp

```diff
@@ -120,16 +153,15 @@ void match_identifiers(TypedAST::DeclarationList* ast, Frontend::CompileTimeEnvi
 
 	for (auto& decl : ast->m_declarations) {
 		auto d = static_cast<TypedAST::Declaration*>(decl.get());
-		if (d->m_value) {
+
+		// this is where we implement let-polymorphism. TODO: refactor.
+		if (d->m_value)
 			match_identifiers(d->m_value.get(), env);
-			d->m_decl_type
-			    = env.m_typechecker.m_core.generalize(d->m_value->m_value_type);
-		} else {
-			// TODO SPEED: dont create a variable and then generalize it, create a
-			// generalized variable directly
-			d->m_decl_type
-			    = env.m_typechecker.m_core.generalize(env.m_typechecker.new_var());
-		}
+
+		MonoId mono = d->m_value ? d->m_value->m_value_type
+			: env.m_typechecker.new_var();
```

Comment:
You changed the ternary operator writing style here hahahaha. I personally like the ? and : being either on the same line or lined up (the latter being your other use case)

Reply:
I like putting things in different lines but this is how it's set up in the .clang-format file

Reply:
Then the other use case did not get formatted